docs/source/getting-started/quickstart-2/index.rst

```diff
@@ -24,24 +24,139 @@ Set up
 
 .. code-block:: bash
 
   export MLFLOW_TRACKING_URI=http://localhost:5000
-  mlflow run -e hyperopt .
 
-The ``hyperopt`` entry point defaults to 12 runs of 32 epochs apiece and should take a few minutes to finish.
+This code defaults to 12 runs of 32 epochs apiece and should take a few minutes to finish.
+
+.. code-block:: python
+
+    import numpy as np
+    import pandas as pd
+    from hyperopt import STATUS_OK, Trials, fmin, hp, tpe
+    from sklearn.metrics import mean_squared_error
+    from sklearn.model_selection import train_test_split
+    from tensorflow.keras.layers import Dense, Lambda
+    from tensorflow.keras.models import Sequential
+    from tensorflow.keras.optimizers import SGD
+
+    import mlflow
+    from mlflow.models import infer_signature
+
+    # Load dataset
+    data = pd.read_csv(
+        "https://raw.githubusercontent.com/mlflow/mlflow/master/tests/datasets/winequality-white.csv",
+        sep=";",
+    )
+
+    # Split the data into training, validation, and test sets
+    train, test = train_test_split(data, test_size=0.25, random_state=42)
+    train_x = train.drop(["quality"], axis=1).values
+    train_y = train[["quality"]].values.ravel()
+    test_x = test.drop(["quality"], axis=1).values
+    test_y = test[["quality"]].values.ravel()
+    train_x, valid_x, train_y, valid_y = train_test_split(
+        train_x, train_y, test_size=0.2, random_state=42
+    )
+
+
+    def train_model(params, train_x, train_y, valid_x, valid_y, test_x, test_y, epochs):
+        # Define model architecture
+        model = Sequential()
+        model.add(
+            Lambda(lambda x: (x - np.mean(train_x, axis=0)) / np.std(train_x, axis=0))
+        )
+        model.add(Dense(64, activation="relu", input_shape=(train_x.shape[1],)))
+        model.add(Dense(1))
+
+        # Compile model
+        model.compile(
+            optimizer=SGD(lr=params["lr"], momentum=params["momentum"]),
+            loss="mean_squared_error",
+        )
+
+        # Train model with MLflow tracking
+        with mlflow.start_run(nested=True):
+            # Fit model
+            model.fit(
+                train_x,
+                train_y,
+                validation_data=(valid_x, valid_y),
+                epochs=epochs,
+                verbose=0,
+            )
+
+            # Evaluate the model
+            predicted_qualities = model.predict(test_x)
+            test_rmse = np.sqrt(mean_squared_error(test_y, predicted_qualities))
+
+            # Log parameters and results
+            mlflow.log_params(params)
+            mlflow.log_metric("test_rmse", test_rmse)
+
+            # Log model
+            signature = infer_signature(train_x, train_y)
+            mlflow.tensorflow.log_model(model, "model", signature=signature)
+
+            return {"loss": test_rmse, "status": STATUS_OK, "model": model}
+
+
+    def objective(params):
```

Comment:
Could we break this code into 3 different sections with a brief explanation in between each to reduce the intimidation factor for users? 
* train_model function with a brief explanation of why we're defining a run context within that function
* objective (super brief explanation of why this needs to be a separate defined function and perhaps a brief explanation of why hyperopt requires a partial function to be defined in this way
* the main parent run context initialization and initiation (with a brief explanation of what the MLflow components of this are doing